### ISSUE TITLE
Feature caching error in shape

### ISSUE THREAD

**Issue by abhijeet3922** [mozilla/DeepSpeech#1772]
- **Running examples on an unmodified clone of the repository**
- **OS Platform and Distribution  - Linux Ubuntu 16.04**
- **TensorFlow version : 1.12.0-rc2**
- **Python version 3.6.6**
- **GPU model and memory - GTX 1080 Ti 11 GB**
- **Exact command to reproduce**:

1. Running this command executes well for 1st time as features are cached.

./DeepSpeech.py --train_files voxforge/voxforge-train.csv --dev_files voxforge/voxforge-dev.csv --test_files voxforge/voxforge-test.csv --train_batch_size 64 --dev_batch_size 64 --test_batch_size 64 --epoch 30 --n_hidden 700 --validation_step 1 --learning_rate 0.0005 --dropout_rate 0.22 --summary_secs 300 --lm_binary_path data/voxforge/lm.binary --lm_trie_path data/voxforge/trie --train_cached_features_path data/voxforge/vox-train --dev_cached_features_path data/voxforge/vox-dev --test_cached_features_path data/voxforge/vox-test 

2. Running the same command again gives below error as cached features could not be loaded properly.

Exception in thread Thread-22:
Traceback (most recent call last):
File "/home/abhijeet/anaconda3/lib/python3.6/threading.py", line 916, in _bootstrap_inner
self.run()
File "/home/abhijeet/anaconda3/lib/python3.6/threading.py", line 864, in run
self._target(*self._args, **self._kwargs)
File "/home/abhijeet/Desktop/dl_projects/DeepSpeech/util/feeding.py", line 143, in _populate_batch_queue
(features.strides[0], features.strides[0], features.strides[1]),
IndexError: tuple index out of range

I tried to debug. I found the error was here in util/preprocess.py.

```
if hdf5_cache_path and os.path.exists(hdf5_cache_path):
        with tables.open_file(hdf5_cache_path, 'r') as file:
            features = file.root.features[:]
            features_len = file.root.features_len[:]
            transcript = file.root.transcript[:]
            transcript_len = file.root.transcript_len[:]

            # features are stored flattened, so reshape into
            # [n_steps, (n_input + 2*n_context*n_input)]
            for i in range(len(features)):
                features[i] = np.reshape(features[i], [features_len[i], -1])

            in_data = list(zip(features, features_len,
                               transcript, transcript_len))
            print('Loaded from cache at', hdf5_cache_path)
return pandas.DataFrame(data=in_data, columns=COLUMNS)
```

When I changed this to 

```
# features are stored flattened, so reshape into
# [n_steps, (n_input + 2*n_context*n_input)]
 for i in range(len(features)):
     features[i] = np.reshape(features[i], [features_len[i]+2*numcontext, numcep, -1])

```
It worked fine and I was able to train the model to achieve 75% accuracywith cached features. Features have to be reshaped to (number of frames, numcep) for each training audio.

Someone can check and confirm if I am correct. 
Correct me if wrong. I pulled the current DeepSpeech repository.

**Comment by reuben** [comment]
Yep, you're right, I have no idea how we haven't run into this in our training runs. I'll fix it.

**Comment by abhijeet3922** [comment]
Thanks @reuben.
This irked me for long time. Somehow I was also procrastinating it to debug. :-)

**Comment by reuben** [comment]
Closed via f5aa47b.
